xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go

```diff
@@ -1317,51 +1120,58 @@ func (s) TestAggregateCluster_SwitchEDSAndDNS(t *testing.T) {
 	// Update DNS resolver with test backend addresses.
 	dnsR.UpdateState(resolver.State{Addresses: addrs[1:]})
 
-	// Make an RPC and ensure that it gets routed to the backend corresponding
-	// to the LOGICAL_DNS cluster.
+	// Ensure that start getting routed to the backend corresponding to the
+	// LOGICAL_DNS cluster.
 	for ; ctx.Err() == nil; <-time.After(defaultTestShortTimeout) {
 		client.EmptyCall(ctx, &testpb.Empty{}, grpc.Peer(peer))
 		if peer.Addr.String() == addrs[1].Addr {
 			break
 		}
 	}
 	if ctx.Err() != nil {
-		t.Fatal("Timeout when waiting for RPCs to be routed to backends in the DNS cluster")
+		t.Fatalf("Timeout when waiting for RPCs to be routed to backend %q in the DNS cluster", addrs[1].Addr)
 	}
 }
 
-// TestAggregateCluster_ErrorsFromEDSAndDNS tests the case where the top-level
-// cluster is an aggregate cluster which resolves to an EDS and DNS cluster.
-// When the EDS cluster resports errors, the test verifies that we switch to the
-// DNS cluster. And once the DNS cluster also returns an error, the test
-// verifies that the error is reported to the caller of the RPC.
-func (s) TestAggregateCluster_ErrorsFromEDSAndDNS(t *testing.T) {
+// TestAggregateCluster_BadEDS_GoodToBadDNS tests the case where the top-level
+// cluster is an aggregate cluster that resolves to an EDS and LOGICAL_DNS
+// cluster. When the EDS request returns a resource that contains no endpoints,
+// the test verifies that we switch to the DNS cluster and can make a successful
+// RPC. At this point when the DNS cluster returns an error, the test verifies
```

Comment:
Nit: the test first asserts no RPC can be made after receiving EDS response with zero endpoints because no update has been pushed through DNS resolver yet.

Reply:
Done.